### PR TITLE
Add MWSC

### PR DESCRIPTION
**Pull request by ghomasHudson**
Adding the [Modified Winograd Schema Challenge](https://github.com/salesforce/decaNLP/blob/master/local_data/schema.txt) dataset which formed part of the [decaNLP](http://decanlp.com/) benchmark. Not sure how much use people would find for it it outside of the benchmark, but it is general purpose.

Code is heavily borrowed from the [decaNLP repo](https://github.com/salesforce/decaNLP/blob/1e9605f246b9e05199b28bde2a2093bc49feeeaa/text/torchtext/datasets/generic.py#L773-L877).

There's a few (possibly overly opinionated) design choices I made:

- I used the train/test/dev split [buried in the decaNLP code](https://github.com/salesforce/decaNLP/blob/1e9605f246b9e05199b28bde2a2093bc49feeeaa/text/torchtext/datasets/generic.py#L852-L855)
- I split out each example into the 2 alternatives. Originally the data uses the format:
    ```
    The city councilmen refused the demonstrators a permit because they [feared/advocated] violence. 
    Who [feared/advocated] violence? 
    councilmen/demonstrators
    ```
    I split into the 2 variants:
    ```
    The city councilmen refused the demonstrators a permit because they feared violence. 
    Who feared violence? 
    councilmen/demonstrators
    
    The city councilmen refused the demonstrators a permit because they advocated violence. 
    Who advocated violence? 
    councilmen/demonstrators
    ```
    I can't see any use for having the options combined into a single example (splitting them is [the way decaNLP processes](https://github.com/salesforce/decaNLP/blob/1e9605f246b9e05199b28bde2a2093bc49feeeaa/text/torchtext/datasets/generic.py#L846-L850)) them. You can't train on both versions with them combined, and splitting the examples later would be a pain to do. I think [winogrande.py](https://github.com/huggingface/nlp/blob/master/datasets/winogrande/winogrande.py) presents the data in this way?

- I've not used the decaNLP framing (appending the options to the question e.g. `Who feared violence? 
 -- councilmen or demonstrators?`) but left it more generic by adding the options as a new key: `"options":["councilmen","demonstrators"]` This should be an easy thing to change using `map` if needed by a specific application.

Dataset is working as-is but if anyone has any thoughts/preferences on the design decisions here I'm definitely open to different choices.

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
Looks good to me

File: datasets/mwsc/mwsc.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        schemas_file = dl_manager.download_and_extract(_DATA_URL)
+
+        if os.path.isdir(schemas_file):
+            # During testing the download manager mock gives us a directory
+            schemas_file = os.path.join(schemas_file,"schema.txt")
```
Comment (lhoestq) [inline datasets/mwsc/mwsc.py:58]: Do you think there is a way to avoid this trick @patrickvonplaten ?

Comment (ghomasHudson) [reply datasets/mwsc/mwsc.py:58]: I borrowed it from https://github.com/huggingface/nlp/blob/7d1526dfeeb29248d832f1073192dbf03ad642da/datasets/tiny_shakespeare/tiny_shakespeare.py#L76 so if there is we can fix it there too.

Comment (lhoestq) [reply datasets/mwsc/mwsc.py:58]: Thanks. We'll keep that in mind for when we re-do a pass over dummy data I think.

**Pull request merged**